roachtest: unskip fuse disk-stall roachtest variant

nicktrav opened the pull request:
**roachtest: unskip fuse disk-stall roachtest variant**

The `fuse` variant of the `disk-stalled` roachtest was skipped in
https://github.com/cockroachdb/cockroach/pull/95865.

Reenable the skipped variant, updating it to make use of our forked
version of `charybdefs`. This fork includes a patch that allows for
specifying a delay time for syscalls, making it possible to simulate a
complete disk stall. Previously, delay times were limited to 50ms, which
meant that the detection time had to be even lower (e.g. 40ms), which
was not representative of how Cockroach is configured in practice.

Fixes https://github.com/cockroachdb/cockroach/issues/95874.

Fixes https://github.com/cockroachdb/cockroach/issues/95815.

Fixes https://github.com/cockroachdb/cockroach/issues/96410.

Fixes https://github.com/cockroachdb/cockroach/issues/95886.

Release note: None.

nicktrav committed:
roachtest: unskip fuse disk-stall roachtest variant

The `fuse` variant of the `disk-stalled` roachtest was skipped in
\#95865.

Re-enable the skipped variant, updating it to make use of our forked
version of `charybdefs`. This fork includes a patch that allows for
specifying a delay time for syscalls, making it possible to simulate a
complete disk stall. Previously, delay times were limited to 50ms, which
meant that the detection time had to be even lower (e.g. 40ms), which
was not representative of how Cockroach is configured in practice.

Allow the roachprod infrastructure to interpolate strings such as
`{store-dir}`, etc. when provided as `ExtraArgs` or the `KeyCmd`. Fix by
expanding expanding all args, rather than just `ExtraArgs`.

Fixes #95874.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96094)
<!-- Reviewable:end -->

nicktrav commented:
```console
HTTP server listening on all network interfaces, port 8080.
test runner logs in: artifacts/_runner-logs/test_runner-1674833993.log
=== RUN   disk-stalled/dmsetup
=== RUN   disk-stalled/fuse/log=false,data=true
=== RUN   disk-stalled/fuse/log=true,data=false
=== RUN   disk-stalled/fuse/log=false,data=false
=== RUN   disk-stalled/fuse/log=true,data=true
=== RUN   disk-stalled/cgroup
--- PASS: disk-stalled/dmsetup (276.14s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/cgroup (272.99s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=true,data=false (696.76s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=false,data=true (751.65s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=false,data=false (836.73s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=true,data=true (979.35s)
no work remaining; runWorker is bailing out...
PASS
```

nicktrav commented:
This should be good for another look. Rebased and fixed up the lingering issues.

Just finished a run from my GCE worker:

```console
=== RUN   disk-stalled/cgroup/read-write/logs-too=false
=== RUN   disk-stalled/fuse/log=false,data=false
=== RUN   disk-stalled/dmsetup
=== RUN   disk-stalled/cgroup/read-write/logs-too=true
=== RUN   disk-stalled/cgroup/write-only/logs-too=true
=== RUN   disk-stalled/fuse/log=true,data=false
=== RUN   disk-stalled/fuse/log=true,data=true
=== RUN   disk-stalled/fuse/log=false,data=true
--- PASS: disk-stalled/cgroup/read-write/logs-too=false (689.18s)
--- PASS: disk-stalled/dmsetup (690.54s)
--- PASS: disk-stalled/cgroup/read-write/logs-too=true (690.69s)
--- PASS: disk-stalled/cgroup/write-only/logs-too=true (689.04s)
no work remaining; runWorker is bailing out...
no work remaining; runWorker is bailing out...
no work remaining; runWorker is bailing out...
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=false,data=false (1126.76s)
--- PASS: disk-stalled/fuse/log=true,data=false (1141.03s)
no work remaining; runWorker is bailing out...
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=true,data=true (1212.45s)
no work remaining; runWorker is bailing out...
--- PASS: disk-stalled/fuse/log=false,data=true (1274.00s)
no work remaining; runWorker is bailing out...
PASS
```

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96094#-:-NNbLtwq101ntLfAp9cC:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @herkolategan, @itsbilal, @nicktrav, and @renatolabs)

___
*[`pkg/cmd/roachtest/tests/disk_stall.go` line 101 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96094#-NNbLcjX48Nz6Ydkz1tr:-NNbLcjX48Nz6Ydkz1ts:b-awp4v1) ([raw file](https://github.com/cockroachdb/cockroach/blob/6ed2fb0f1e98b0bdbf6d0f100bef2c1bf5dbf6de/pkg/cmd/roachtest/tests/disk_stall.go#L101)):*
> ```Go
> 		"--store", s.DataDir(),
> 		"--log", fmt.Sprintf(`{sinks: {stderr: {filter: INFO}}, file-defaults: {dir: "%s"}}`, s.LogDir()),
> 	}
> ```

thanks for persevering through all the roachprod log and store directory drudgery 


<!-- Sent from Reviewable.io -->

nicktrav commented:
Going to land this, as it addresses a bunch of open issues.

Test-Eng folks - happy to follow up separately on any feedback on the roachprod changes.

TFTR!

bors r=jbowens
